packages/context/test/unit/context.ts

```diff
@@ -6,6 +6,54 @@
 import {expect} from '@loopback/testlab';
 import {Context, Binding, BindingScope, BindingType, isPromise} from '../..';
 
+/**
+ * Create a subclass of context so that we can access parents and registry
+ * for assertions
+ */
+class TestContext extends Context {
+  get parentList() {
+    return this.parents;
+  }
+  get bindingMap() {
+    const map = new Map(this.registry);
+    return map;
+  }
+}
+
+describe('Context constructor', () => {
+  it('generates a unique name', () => {
+    const ctx = new Context();
+    expect(ctx.name).to.match(
+      /^[0-9A-F]{8}-[0-9A-F]{4}-[1][0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i,
```

Comment:
This check does not really guarantee that names are unique, it allows the context to always set the name to the same uuid.

A more correct test is to create two contexts and verify that their names are different.

```ts
const ctx1 = new Context();
const ctx2 = new Context();

expect(ctx1.name).to.not.equal(ctx2.name);
```

I am ok with keeping your current check too.